replication-api-impl/src/main/java/org/codice/ditto/replication/api/impl/HeartbeaterAndVerifierImpl.java

```diff
@@ -73,23 +161,404 @@ public void cleanUp() {
 
   @Override
   public void verify(ReplicationSite site) {
+    if (sitesToSkip.contains(site.getId())) {
+      LOGGER.debug(
+          "Heartbeat to Site {} skipped since last request was a bad request. Restart the system to trigger a retry of this site.",
+          site.getUrl());
+      return;
+    }
+
+    if (site.getVerifiedUrl() == null) {
+      doHeartbeat(site, site.getUrl());
+    }
+  }
+
+  private HeartbeatResult doHeartbeat(ReplicationSite site, String url) {
+    int numRedirects = 0;
+    HeartbeatResult heartbeatResult = doDiscovery(url, site.getId());
+    do {
+
+      boolean redirected = handleHeartbeatResult(heartbeatResult, site);
+      if (redirected) {
+        url = heartbeatResult.getRedirectUrl();
+        numRedirects++;
+
+        if (heartbeatResult.isPermRedirect()) {
+          site.setRemoteManaged(true);
+          site.setVerifiedUrl(url);
```

Comment:
This will set the verified URL as the perm redirect URL every time, but I just realized technically this is not a verified URL until we test the redirect URL so probably cannot save it every time.

Reply:
Because it is a permanent redirect we have to save it. What we cannot do is assumed it is remotely managed yet.